### PR TITLE
feat: add web login methods

### PR DESCRIPTION
**Pull request by chad3814**
This lets you either present a url for a user to go to, or a url to send them too (if in a webserver context) and then get a token from Plex.

I created an example usage in the `example` directory

### CODE REVIEW

File: src/myplex.ts [hunk 1]
```diff
@@ -20,6 +20,94 @@ import { MediaContainer } from './util.js';
...
+      } catch (err) {
+        if ((err as Error).message.includes('aborted')) {
+          continue;
+        }
+
+        if ((err as Error).message.includes('429')) {
+          // eslint-disable-next-line no-await-in-loop
+          await sleep(recheckMs);
+          continue;
+        }
```
Comment (scttcper) [inline src/myplex.ts]: i think this sort of thing should already be handled by ofetch's retry? https://github.com/unjs/ofetch#%EF%B8%8F-auto-retry

Comment (chad3814) [reply src/myplex.ts]: oh yeah, I did do that there, afterwards

**Comment by chad3814** [comment]
basically I want to use this in a web server and don't want to freak people out by asking for their credentials :)

**Comment by scttcper** [comment]
i think this makes sense, i'm just looking at what they python library does since this one is mostly based on it https://github.com/pkkid/python-plexapi/blob/d5604670d5b7f7ff9c882e41f6882454a9b4e4e0/plexapi/myplex.py#L1720-L1743

**Comment by chad3814** [comment]
> i think this makes sense, i'm just looking at what they python library does since this one is mostly based on it https://github.com/pkkid/python-plexapi/blob/d5604670d5b7f7ff9c882e41f6882454a9b4e4e0/plexapi/myplex.py#L1720-L1743

If you want, I can rename stuff to line up with the python code, but it looks like it's essentially the same (I'm not a big python guy).

**Comment by scttcper** [comment]
@chad3814 its okay, just a heads up it might change to match theirs eventually

**Pull request merged**